### PR TITLE
fix: ensure implicit config options load automatically

### PR DESCRIPTION
**Pull request by lsh4711**
**Issue**

Since version `6.3.14-dev.57` of `@mikro-orm/core`, modifications in the `ConfigurationLoader` related code have led to the `paths is not iterable` error if a config is not explicitly provided to `MikroOrmModule.forRoot()`.

**Changes**

- Adjusted to align with the new `getConfiguration()` signature.
- Prefer `configPathsFromArg()` when auto-loading options(but this has become `deprecated`).
- Added test code to ensure options auto-loading.

### CODE REVIEW

**Comment by B4nan** [comment]
cc @boenrobot maybe we should make the signature backwards compatible so we don't break this integration

**Comment by B4nan** [comment]
I would rather go with what I said above, we don't need to change the order of params in that method (only add the new params to the end), that's a small price to pay for proper back compatibility on internal API. Or even detect old vs new signature. You never know if some other integration modules that we don't maintain use things like this, better to stay safe. We can clean things up with v7.

> This would also enable the lack of duplicating the deprecation notice... it could be a new major of this package that simply removes support for it.

What I want would mean no changes in this package. I definitely don't want to do a major bump to misalign the versions between it and the ORM itself.

**Comment by B4nan** [comment]
> I think perhaps this BC can be achieved by making ConfigLoader.getConfiguration() be overloaded, and based on whether the first arg is boolean, trigger the old behavior. The boolean signature can be marked deprecated in js docs to give 3rd party integrations a notice.

Yep, that's what I meant when I said we could detect the old vs new signatures.

> I can make a PR for this first thing in the morning (around 12-13 hours from now...).

Sounds good, no rush, we can leave the release for tomorrow evening or later, not a huge deal. Thanks!

**Comment by B4nan** [comment]
@lsh4711 please test it with latest dev version (`6.3.14-dev.70`), should be now working thanks to https://github.com/mikro-orm/mikro-orm/pull/6213

**Comment by lsh4711** [comment]
In `6.3.14-dev.70`, the path issue was resolved, but it still didn't work, so I've been investigating the cause.
Here's the [reproduction](https://github.com/lsh4711/reproduction/tree/mikro-orm/nestjs).
I belatedly found out that the issue does not reproduce in the test when using mocked dynamic import.

**Comment by B4nan** [comment]
> But this can be easily fixed (without a version bump in the deps) by changing that line to

As I said, I don't want to fix it here, as that means people can end up with typeerrors because of updating the ORM and not this adapter. This needs to be fixed in the ORM itself before we release v6.4

**Comment by B4nan** [comment]
@lsh4711 I guess you closing this means its working correctly now? :]

**Comment by lsh4711** [comment]
Oh.. I thought the issue was resolved by now and closed this PR without checking, which was a mistake and the problem still hasn't been fixed.
As mentioned above, the same issue is still occurring both in the reproduction and on the actual server. (in `6.3.14-dev.73`)

Based on my investigation, the solution I found is to use the call to ConfigurationLoader.commonJSCompat() to maintain compatibility with the behavior of the previous version.
In other words, since this code is missing, the config object that has been wrapped once more ({ default: config, ..else config in same config file }) is being dynamically imported, so the config isn't being interpreted correctly, which seems to be why the issue persists.

**Comment by B4nan** [comment]
Ok, lets reopen till its actually fixed.

**Comment by lsh4711** [comment]
Great, it works perfectly now. :)
Thanks to both B4nan and boenrobot!
